app/concerns/assignable_types.rb

```diff
@@ -18,7 +18,10 @@ module ClassMethods
     def load_types_in(module_name, my_name = module_name.singularize)
       const_set(:MODULE_NAME, module_name)
       const_set(:BASE_CLASS_NAME, my_name)
-      const_set(:TYPES, Dir[Rails.root.join("app", "models", module_name.underscore, "*.rb")].map { |path| module_name + "::" + File.basename(path, ".rb").camelize })
+      types = module_name.constantize.constants
```

Comment:
I'm afraid this may not work in development and test environments where files are lazy-loaded.  Can you confirm this works even in development mode? (e.g. `bundle exec rails r 'puts Agent::TYPES'`)